### ISSUE TITLE
[Question] migration to ubuntu 20.04/fresh os

### ISSUE THREAD

**Issue by dumprop** [caprover/caprover#1322]
Support for Ubuntu 18.04 will end in April 2023. 
So in general, servers with a caprover may have problems getting security updates after a year
I looked at the tickets that were specified in the documentation and seems that they have been resolved
![Screenshot_20220202_231124](https://user-images.githubusercontent.com/34004367/152229678-e1dfd5a7-83aa-437e-bc37-1a1faff513ae.png)
![Screenshot_20220202_231247](https://user-images.githubusercontent.com/34004367/152229853-286c0410-4641-4d46-9c64-f72f42221d3c.png)
Is anyone testing the installation on a newer OS?
I also noticed that Ubuntu has a free subscription to receive updates for another 5 years for personal use and up to 3 servers (as workaround)

**Comment by githubsaturn** [comment]
@dumprop - drop this question in the Slack channel for better reach... 

The only reason that Ubuntu 18 is the recommended stack is because of those issues. If they are resolved, there is no reason not to upgrade.

**Comment by maietta** [comment]
He @dumprop, the screen shots you provided didn't seem to come with any link to the issues you screenshotted. As a result, it makes it difficult for people like me to quickly take a peek at the issues opened that related to your issue here. I have provided the links for others like myself to refer to in the future.

Upgrade to 20.10 breaks swarm network
https://github.com/moby/moby/issues/41775

and 

All published services within a docker swarm are unreachable while containers deployed normally work fine.
https://github.com/moby/moby/issues/41825

It appears that for both the first and second issues, some resolution has been found but I have yet to confirm this myself.

Hoping this is the case because I would really like to setup a Home Lab using a Pi 4 so I can run Caprover. Currently, the Pi 4 does not seem to be supported by 18.04 LTS, the currently and officially recommended distro for a stable Caprover.

**Comment by maietta** [comment]
Checking in to report that Caprover is running nicely on Ubuntu Server 20.04 LTS. Running on a Dell XPS 9560 with an SSD storage device and 32GB ram. I'm operating on a laptop for ultra-portability. Enabled SSL support and real domains by creating a VPN Gateway with Wireguard and Haproxy on a cheap VPS as the public node, then the laptop creating a VPN with Wireguard as a peer. Caprover was setup using the VPS static IP as the public node. I'll be doing a full write-up on this.

**Comment by githubsaturn** [comment]
Great to hear!
It’d also be good to test a multi node setup as that appears to be the biggest issue.

**Comment by maietta** [comment]
> Great to hear! It’d also be good to test a multi node setup as that appears to be the biggest issue.

Indeed. I will be working to test this hopefully sometime next week. (Finding time has been difficult). I'll report back whatever I discover.

**Comment by lolrenx** [comment]
Any news on the multi nodes support for Ubuntu 20.04 LTS ?

**Comment by githubsaturn** [comment]
CapRover already support 20.04 - these issues are Docker related.

**Comment by Tam2** [comment]
I've got cluster working on Ubuntu 22.04 too, no issues so far

**Comment by maietta** [comment]
Update from me: Initial testing of a docker multi-node swarm looks good with 20.04 here too!

**Comment by githubsaturn** [comment]
Thanks @maietta ! If you all feel good about Ubuntu 20, feel free to open a PR and update the recommended stack here:
https://caprover.com/docs/get-started.html#b2-server-specs

**Comment by maietta** [comment]
> 

Will do!
